### PR TITLE
[Workflow] Add mypy check to pre-commit

### PR DESCRIPTION
**Pull request by dandansamax**
## Description

Add Check mypy to pre-commit hooks.

## Motivation and Context

Why is this change required? What problem does it solve?
If it fixes an open issue, please link to the issue here.
You can use the syntax `close #15213` if this solves the issue #15213

- [ ] I have raised an issue to propose this change ([required](https://github.com/camel-ai/camel/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of example)

## Implemented Tasks

- [ ] Subtask 1
- [ ] Subtask 2
- [ ] Subtask 3

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/camel-ai/camel/blob/master/CONTRIBUTING.md) guide. (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -17,6 +17,15 @@ repos:
     hooks:
       - id: flake8
         name: Check PEP8
+  
+  - repo: local
+    hooks:
+      - id: mypy
+        name: Check mypy
+        entry: "./scripts/run-mypy"
```
Comment (lightaime) [inline .pre-commit-config.yaml:25]: Is it necessary to create a script? Can we just do:
```suggestion
        entry: mypy --namespace-packages --install-types --non-interactive -p camel -p test -p apps
```

Comment (dandansamax) [reply .pre-commit-config.yaml]: Truely make sense.

**Review by lightaime — APPROVED** [review]
LGTM!

**Comment by Wendong-Fan** [comment]
hey @dandansamax @lightaime ,
After adding this mypy check we have many errors when doing commits like the screenshot below, maybe we also to fix all these errors caused by our early implemented features?

Screenshot 2023-11-19 at 01 33 27

**Pull request merged**